code/game/mecha/equipment/weapons/mecha_energy.dm

```diff
@@ -0,0 +1,119 @@
+/obj/item/mecha_parts/mecha_equipment/weapon/energy
+	name = "general energy weapon"
+	auto_rearm = 1
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser
+	equip_cooldown = 8
+	name = "mounted laser carbine"
+	desc = "A weapon for combat exosuits. The CH-PS \"Immolator\" laser shoots basic lasers."
+	icon_state = "mecha_laser"
+	energy_drain = 30
+	projectile = /obj/item/projectile/beam/midlaser
+	fire_sound = 'sound/weapons/Laser.ogg'
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/gatling
+	name = "mounted gatling laser"
+	desc = "A weapon for combat exosuits. Shoots a burst of basic lasers."
+	energy_drain = 45
+	projectiles_per_shot = 8
+	deviation = 0.7
+	fire_time = 5
+	projectiles_per_shot = 5
+	deviation = 0.5
+	fire_cooldown = 10
+	fire_time = 10
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/riggedlaser
+	equip_cooldown = 30
+	name = "jury-rigged welder-laser"
+	desc = "While not regulation, this inefficient weapon can be attached to working exo-suits in desperate, or malicious, times."
+	icon_state = "mecha_laser"
+	energy_drain = 80
+	projectile = /obj/item/projectile/beam
+	fire_sound = 'sound/weapons/Laser.ogg'
+	required_type = list(/obj/mecha/combat, /obj/mecha/working)
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy
+	equip_cooldown = 15
+	name = "mounted laser cannon"
+	desc = "A weapon for combat exosuits. The CH-LC \"Solaris\" laser cannon is designated to shoot heavy lasers."
+	icon_state = "mecha_laser"
+	energy_drain = 60
+	projectile = /obj/item/projectile/beam/heavylaser
+	fire_sound = 'sound/weapons/lasercannonfire.ogg'
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/ion
+	equip_cooldown = 40
+	name = "heavy ion cannon"
+	desc = "A weapon for combat exosuits. The mkIV ion heavy cannon shoots ion bolts designated to disable mechanical threats."
+	icon_state = "mecha_ion"
+	energy_drain = 120
+	projectile = /obj/item/projectile/ion/heavy
+	fire_sound = 'sound/weapons/Laser.ogg'
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/pulse
+	equip_cooldown = 30
+	name = "heavy pulse cannon"
+	icon_state = "mecha_pulse"
+	desc = "A weapon for combat exosuits. The eZ-13 mk2 heavy pulse rifle shoots powerful pulse-based beams, capable of destroying structures."
+	energy_drain = 120
+	origin_tech = list(TECH_MATERIAL = 3, TECH_COMBAT = 6, TECH_POWER = 4)
+	projectile = /obj/item/projectile/beam/pulse/heavy
+	fire_sound = 'sound/weapons/marauder.ogg'
+
+/obj/item/projectile/beam/pulse/heavy
```

Comment:
Please move that to the projectile definitions in code/modules/projecties/projectile/beams.dm